src/react-hooks/buildHooks.ts

```diff
@@ -262,6 +322,25 @@ export function buildHooks<Definitions extends EndpointDefinitions>({
     return {
       useQueryState,
       useQuerySubscription,
+      useLazyQuerySubscription,
+      useLazyQuery(options) {
+        const [providedArgs, setProvidedArgs] = useState(null);
+        const [trigger, lastPromise] = useLazyQuerySubscription(options);
+        const queryStateResults = useQueryState(providedArgs, {
+          ...options,
+          skip: providedArgs === null,
+        });
+        const triggerQuery = useCallback(
+          function (args: any) {
+            setProvidedArgs(args);
+            trigger(args);
+          },
+          [trigger]
+        );
+
+        const info = useMemo(() => ({ lastArgs: lastPromise?.arg }), [lastPromise]);
+        return useMemo(() => [triggerQuery, queryStateResults, info], [triggerQuery, queryStateResults, info]);
```

Comment:
```suggestion
        const [trigger, lastPromise] = useLazyQuerySubscription(options);
        const queryStateResults = useQueryState(lastPromise.arg, {
          ...options,
          skip: !lastPromise,
        });

        const info = useMemo(() => ({ lastArgs: lastPromise?.arg }), [lastPromise]);
        return useMemo(() => [trigger, queryStateResults, info], [trigger, queryStateResults, info]);
```
Maybe doing it like this would also resolve that problem?

Reply:
💯